### ISSUE TITLE
Write Firmware -> Directory is missing! 

### ISSUE THREAD

**Issue by jj-uk** [opentx/opentx#7354]
Taranis X9D+ (2015)

I've just updated to 2.3.5, and formatted the SDCard and loaded "sdcard-212x64-2.3V0025" contents to the SD Card.

From the bootloader (which is version 2.0.15), if i go to Write Frimware, or Restore EEPROM, i see "Directory is empty!".

I have reloaded the previous firmware 2.2.3 back to the radio, and restored the original SD card contents for this version to the SD card, and i've tried a new SD card, but nothing is working.

I flashed a receiver with the radio yesterday, but today, after the update to 2.3.5, I get "Directory is empty!" error.

Can someone suggest a solution? What causes this error? From the SD card viewer in the radio menu, i can browse the SD card files.

**Comment by jj-uk** [comment]
I've managed to update the bootloader by dropping the 2.3.5 firmware in to the FIRMWARE folder on the USB card and then executing the bootloader update from the SD Card browser (not from the bootloader).

Once the bootloader was updated and power-cycled, I can again access the folders.

**Comment by kilrah** [comment]
Directory name changed in 2.2.

**Comment by gusemma** [comment]
My X9D+ will only open Bootloader 2.0.9 with Write Firmware, Restore EEPROM, and Exit. From there I get "Directory is missing" and that's all, no other functions. What must I do?

**Comment by 3djc** [comment]
Update bootloader.

**Comment by mraiman001** [comment]
I had this same issue, I did update bootloader, but still was getting "directory is missing"
What solved the problem for me was changing the file location on my sd card.  I initially copied the sd card contents to a separate folder on the sd card.  Apparently, it didn't like that.  I corrected this and copied the sd card contents to the card without using any subfolders and presto, it worked.  Also, as recommended by TweetFPV, I shortened the file name to 8 characters.
